api/v1beta1/kafkacluster_types.go

```diff
@@ -441,11 +449,23 @@ type InternalListenerConfig struct {
 type CommonListenerSpec struct {
 	// +kubebuilder:validation:Enum=ssl;plaintext;sasl_ssl;sasl_plaintext
 	Type SecurityProtocol `json:"type"`
+	// ServerSSLCertSecret is a reference to that secret where custom server SSL certificate can be provided.
+	// Secret has to contain the keystore and truststore in jks and the password for them in base64 encoded format.
+	// Data fields must be: keystore.jks, truststore.jks, password
+	// SecurityProtocol has to be set to ssl.
+	ServerSSLCertSecret *corev1.LocalObjectReference `json:"serverSSLCertSecret,omitempty"`
 	// +kubebuilder:validation:Pattern=^[a-z0-9\-]+
 	Name          string `json:"name"`
 	ContainerPort int32  `json:"containerPort"`
 }
 
+func (c CommonListenerSpec) GetServerSSLCertSecretName() string {
```

Comment:
This should be *CommonListenerSpec do not use two separate receiver for this struct.